sqlobject/src/main/java/org/jdbi/v3/sqlobject/UpdateHandler.java

```diff
@@ -37,11 +37,16 @@
         boolean isGetGeneratedKeys = method.isAnnotationPresent(GetGeneratedKeys.class);
 
         Type returnType = GenericTypes.resolveType(method.getGenericReturnType(), sqlObjectType);
-        if (!isGetGeneratedKeys && returnTypeIsInvalid(method.getReturnType()) ) {
-            throw new UnableToCreateSqlObjectException(invalidReturnTypeMessage(method, returnType));
+        if (!isGetGeneratedKeys) {
```

Comment:
If you invert this condition (and swap the if/else blocks) then inner if/else chain below can make one if/else chain instead of a nested one.